nomad/resource_job.go

```diff
@@ -54,16 +51,13 @@ func resourceJobRegister(d *schema.ResourceData, meta interface{}) error {
 	jobspecRaw := d.Get("jobspec").(string)
 
 	// Parse it
-	jobspecStruct, err := jobspec.Parse(strings.NewReader(jobspecRaw))
+	job, err := jobspec.Parse(strings.NewReader(jobspecRaw))
 	if err != nil {
 		return fmt.Errorf("error parsing jobspec: %s", err)
 	}
 
-	// Initialize and validate
-	jobspecStruct.Canonicalize()
-	if err := jobspecStruct.Validate(); err != nil {
```

Comment:
I understand the validation is now part of `Parse`, right? So it's just implicit, rather than explicit? 🤔

Reply:
@radeksimko It is not validated as part of `Parse()`. The job is validated during Register. If it is invalid and error will be returned. If you want early validation,  you would have to call validate before register: https://github.com/hashicorp/nomad/blob/master/api/jobs.go#L42

Reply:
Understood. So the user practically won't see any difference, because it's validated in CRUD - in either case during `apply`. If we wanted to do early validation we'd do it in `ValidateFunc` (which runs during `plan`).

Still a little bit confusing to see this being removed as part of an upgrade, but it doesn't do any harm, I guess.

Reply:
It was removed because `jobspec.Parse` returns `*api.Job` which does not have a `Validate` function.  That was the simple solution.  Based on the conversation I don't believe there's much value in performing additional validation, but I'm happy to if that's what you'd like.